### ISSUE TITLE
Frozen configs

### ISSUE THREAD

**Issue by playpauseandstop** [hynek/environ-config#11]
Hi,

For some reason I just found out this library and loved it. It allows me to setup `attr.s` settings classes in way more better fashion than before!

However I think one of most important config characteristics is immutability, while by default `@environ.config` creates mutable `attr.s` class,

```python
>>> import environ
>>> @environ.config(prefix="")
... class Settings:
...     LEVEL: str = environ.var()
... 
>>> settings = environ.to_config(Settings)
>>> settings.LEVEL
'dev'
>>> settings.LEVEL = 'prod'
>>> settings
Settings(LEVEL='prod')
```

With that in mind I believe there should be a simple way of passing `frozen=True` to [attr.s](https://github.com/hynek/environ-config/blob/master/src/environ/_environ_config.py#L61) call, which resulted in creation of immutable config instance.

If you're ok with the idea, I can provide the PR to satisfy that and allow `@environ.config` be immutable.

Thanks again!

**Comment by hynek** [comment]
Sure, sounds good to me!

**Comment by hynek** [comment]
Fixed by #12
